### ISSUE TITLE
Support Configurable timeouts for GraphQL Resolver

### ISSUE THREAD

**Issue by afro-coder** [solo-io/gloo#7950]
### Version

1.13.x (latest stable)

### Is your feature request related to a problem? Please describe.

Currently we have a customer using GraphQL which includes multiple API calls however the total request takes greater than 1 second, which causes a 
```
rest resolver got response status 504 from the upstream, body: 'upstream request timeout'
```

Currently there is no way to pass the routeOption for timeout back to the envoy config that is generated.

### Describe the solution you'd like

The Ability to specify a timeout for the total duration of the request or per call 

### Describe alternatives you've considered

I haven't found any alternative for this currently discussed with the team and it relates to solo-io/gloo-mesh-enterprise#7144

**Comment by jbohanon** [comment]
/kick https://console.cloud.google.com/cloud-build/builds/30c9e884-8990-479d-86a9-4d2cd052e4cd;step=0?project=gloo-ee

**Comment by jbohanon** [comment]
This work is released in Gloo Edge EE
- 1.14.2
- 1.15.0-beta1

**Comment by afro-coder** [comment]
Hi,
Would it be possible to backport this to GE 1.13

Thanks.

**Comment by jbohanon** [comment]
Will be released in Gloo Edge EE 1.13.18
